src/function/scalar/string/hex.cpp

```diff
@@ -4,9 +4,41 @@
 #include "duckdb/common/vector_operations/unary_executor.hpp"
 #include "duckdb/common/vector_operations/vector_operations.hpp"
 #include "duckdb/function/scalar/string_functions.hpp"
+#include "duckdb/storage/compression/chimp/algorithm/chimp_utils.hpp"
 
 namespace duckdb {
 
+static void WriteHexBytes(uint64_t x, char *&output, idx_t leading_zero, idx_t type_size) {
+	D_ASSERT(type_size >= leading_zero);
+
+	idx_t offset = type_size - ((leading_zero / 4) * 4);
+
+	for (; offset >= 4; offset -= 4) {
+		uint8_t byte = (x >> (offset - 4)) & 0x0F;
+		*output = Blob::HEX_TABLE[byte];
+		output++;
+	}
+}
+
+static void WriteBinBytes(uint64_t x, char *&output, idx_t leading_zero, idx_t type_size) {
+	D_ASSERT(type_size >= leading_zero);
+	idx_t offset = type_size - leading_zero;
+	idx_t remainder = 8 - (leading_zero % 8);
+	for (idx_t i = 0; i < remainder; ++i) {
+		*output = ((x >> (offset - 1)) & 0x01) + '0';
```

Comment:
The assertion says type_size >**=** leading_zero
so `type_size - leading_zero` could result in `offset` being 0, which results in `remainder` being 0 as well
So it wouldn't enter the loop, I think this is all correct, but can we add a `D_ASSERT(offset >= 1)` just so we're confident it's never doing `x >> -1` because that would cause UB.